tests/drivers/i2s/i2s_additional/src/main.c

```diff
@@ -0,0 +1,862 @@
+
+#define NUM_BLOCKS 20
+#define TIMEOUT 1000
+
+#define SAMPLES_COUNT 64
+
+/* The data_l represent a sine wave */
+static int16_t data_l[SAMPLES_COUNT] = {
+	  3211,   6392,   9511,  12539,  15446,  18204,  20787,  23169,
+	 25329,  27244,  28897,  30272,  31356,  32137,  32609,  32767,
+	 32609,  32137,  31356,  30272,  28897,  27244,  25329,  23169,
+	 20787,  18204,  15446,  12539,   9511,   6392,   3211,      0,
+	 -3212,  -6393,  -9512, -12540, -15447, -18205, -20788, -23170,
+	-25330, -27245, -28898, -30273, -31357, -32138, -32610, -32767,
+	-32610, -32138, -31357, -30273, -28898, -27245, -25330, -23170,
+	-20788, -18205, -15447, -12540,  -9512,  -6393,  -3212,     -1,
+};
+
+/* The data_r represent a sine wave shifted by 90 deg to data_l sine wave */
+static int16_t data_r[SAMPLES_COUNT] = {
+	 32609,  32137,  31356,  30272,  28897,  27244,  25329,  23169,
+	 20787,  18204,  15446,  12539,   9511,   6392,   3211,      0,
+	 -3212,  -6393,  -9512, -12540, -15447, -18205, -20788, -23170,
+	-25330, -27245, -28898, -30273, -31357, -32138, -32610, -32767,
+	-32610, -32138, -31357, -30273, -28898, -27245, -25330, -23170,
+	-20788, -18205, -15447, -12540,  -9512,  -6393,  -3212,     -1,
+	  3211,   6392,   9511,  12539,  15446,  18204,  20787,  23169,
+	 25329,  27244,  28897,  30272,  31356,  32137,  32609,  32767,
+};
+
+#define BLOCK_SIZE (2 * sizeof(data_l))
+
+#ifdef CONFIG_NOCACHE_MEMORY
+	#define MEM_SLAB_CACHE_ATTR __nocache
+#else
+	#define MEM_SLAB_CACHE_ATTR
+#endif /* CONFIG_NOCACHE_MEMORY */
+
+/*
+ * NUM_BLOCKS is the number of blocks used by the test. Some of the drivers,
+ * permanently keep ownership of a few RX buffers. Add a two more
+ * RX blocks to satisfy this requirement
+ */
+static char MEM_SLAB_CACHE_ATTR __aligned(WB_UP(32))
+	_k_mem_slab_buf_rx_0_mem_slab[(NUM_BLOCKS + 2) * WB_UP(BLOCK_SIZE)];
+STRUCT_SECTION_ITERABLE(k_mem_slab, rx_0_mem_slab) =
+	Z_MEM_SLAB_INITIALIZER(rx_0_mem_slab, _k_mem_slab_buf_rx_0_mem_slab,
+				WB_UP(BLOCK_SIZE), NUM_BLOCKS + 2);
+
+static char MEM_SLAB_CACHE_ATTR __aligned(WB_UP(32))
+	_k_mem_slab_buf_tx_0_mem_slab[(NUM_BLOCKS) * WB_UP(BLOCK_SIZE)];
+STRUCT_SECTION_ITERABLE(k_mem_slab, tx_0_mem_slab) =
+	Z_MEM_SLAB_INITIALIZER(tx_0_mem_slab, _k_mem_slab_buf_tx_0_mem_slab,
+				WB_UP(BLOCK_SIZE), NUM_BLOCKS);
+
+static const struct device *dev_i2s;
+
+static const struct i2s_config default_i2s_cfg = {
+	.word_size = WORD_SIZE,
+	.channels = NUMBER_OF_CHANNELS,
+#if !defined(CONFIG_I2S_TEST_DATA_FORMAT_I2S_UNSUPPORTED)
+	.format = I2S_FMT_DATA_FORMAT_I2S,
+#elif !defined(I2S_TEST_DATA_FORMAT_LEFT_JUSTIFIED_UNSUPPORTED)
+	.format = I2S_FMT_DATA_FORMAT_LEFT_JUSTIFIED,
+#elif !defined(I2S_TEST_DATA_FORMAT_RIGHT_JUSTIFIED_UNSUPPORTED)
+	.format = I2S_FMT_DATA_FORMAT_RIGHT_JUSTIFIED,
+#elif !defined(I2S_TEST_DATA_FORMAT_PCM_LONG_UNSUPPORTED)
+	.format = I2S_FMT_DATA_FORMAT_PCM_LONG,
+#elif !defined(I2S_TEST_DATA_FORMAT_PCM_SHORT_UNSUPPORTED)
+	.format = I2S_FMT_DATA_FORMAT_PCM_SHORT,
```

Comment:
`CONFIG_` missing in these data format options. But will there actually be a case where an I2S driver would not support the standard I2S data format?